magento-research/pwa-studio
# dev: Adds a withLogger util function for useReducer in peregrine

## Description

This adds a util, `withLogger` that basically behaves just like our current `middleware/log` does for redux but for `useReducer`.

## Related Issue





Closes #ISSUE_NUMBER.

## Verification Steps

1. Run the app, open search and type a few letters. You should see a log from the new reducer wrapper.
2. Emit a toast (set offline) and see the log for toasts.
2. Run `yarn run storybook:venia` and go to Toasts.
3. Inspect the console and see that the actions are logged along with state updates.

## Screenshots / Screen Captures (if appropriate)
useToasts:
[![Image from Gyazo](https://i.gyazo.com/85618394b86beb0c493cfa3f9d2cf4ee.png)](https://gyazo.com/85618394b86beb0c493cfa3f9d2cf4ee)

useQueryResult:
[![Image from Gyazo](https://i.gyazo.com/c294a2ce3af72228fff323f17341b7fd.png)](https://gyazo.com/c294a2ce3af72228fff323f17341b7fd)
## Proposed Labels for Change Type/Package




## Checklist:
